src/test/java/com/epam/poc/homePage/CategoriesListSectionTest.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.poc.homePage;
+
+import com.epam.poc.commons.BaseTest;
+import com.epam.poc.pageObjects.homepage.CategoriesListSectionObject;
+import com.epam.poc.pageObjects.homepage.HomePageObject;
+import com.epam.poc.utilities.listeners.TestListener;
+import io.qameta.allure.Description;
+import io.qameta.allure.Epic;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+import static com.epam.poc.pageUIs.homepage.CategoriesListSectionUI.*;
+import static com.epam.poc.pageUIs.homepage.CategoriesTilesSectionUI.MAIN_CATEGORY_CSS;
+
+@Listeners({TestListener.class})
+@Epic("Regression test")
+@Feature("HomePage")
+public class CategoriesListSectionTest extends BaseTest {
+    private CategoriesListSectionObject categoriesListSection;
+
+    @BeforeClass
+    public void beforeClass() {
+        categoriesListSection = new CategoriesListSectionObject(driver);
+        homePage = new HomePageObject(driver);
+
+        homePage.closeHomePagePopup();
+    }
+
+    @Test
+    @Description("Test description: Verify random category in List Categories in Home Page")
+    @Story("Categories list section")
+    @Parameters({"pageUrl"})
+    public void verifyCategoriesLink() {
+        //click random category and get name
+        String expectedCategory = categoriesListSection.clickRandomCategoryAndGetName(By.xpath(MAIN_CATEGORY_XPATH), "main-category")
+                .toLowerCase();
+
+        WebElement searchField = homePage.getElement(driver, By.xpath(INPUT_SEARCH_FIELD_XPATH));
+        categoriesListSection.waitForElementContainsText(driver, searchField, expectedCategory);
```

Comment:
suggest moving lines 39 to 43 and line 49 to PageObject.
in the test class, should avoid calling out the PageUI elements.